x-pack/plugins/stack_alerts/public/rule_types/es_query/expression/esql_query_expression.tsx

```diff
@@ -0,0 +1,255 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useState, Fragment, useEffect, useCallback } from 'react';
+import { get } from 'lodash';
+import { FormattedMessage } from '@kbn/i18n-react';
+import {
+  EuiFieldNumber,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiFormRow,
+  EuiSelect,
+  EuiSpacer,
+  EuiTitle,
+} from '@elastic/eui';
+import { getFields, RuleTypeParamsExpressionProps } from '@kbn/triggers-actions-ui-plugin/public';
+import { TextBasedLangEditor } from '@kbn/text-based-languages/public';
+import { fetchFieldsFromESQL } from '@kbn/text-based-editor';
+import { AggregateQuery, getIndexPatternFromESQLQuery } from '@kbn/es-query';
+import { parseDuration } from '@kbn/alerting-plugin/common';
+import {
+  firstFieldOption,
+  getTimeFieldOptions,
+  getTimeOptions,
+  parseAggregationResults,
+} from '@kbn/triggers-actions-ui-plugin/public/common';
+import { EsQueryRuleParams, EsQueryRuleMetaData, SearchType } from '../types';
+import { DEFAULT_VALUES } from '../constants';
+import { useTriggerUiActionServices } from '../util';
+import { hasExpressionValidationErrors } from '../validation';
+import { TestQueryRow } from '../test_query_row';
+import { rowToDocument, toEsQueryHits, transformDatatableToEsqlTable } from '../../../../common';
+
+export const EsqlQueryExpression: React.FC<
+  RuleTypeParamsExpressionProps<EsQueryRuleParams<SearchType.esqlQuery>, EsQueryRuleMetaData>
+> = ({ ruleParams, setRuleParams, setRuleProperty, errors }) => {
+  const { expressions, http } = useTriggerUiActionServices();
+  const { esqlQuery, timeWindowSize, timeWindowUnit, timeField } = ruleParams;
+
+  const [currentRuleParams, setCurrentRuleParams] = useState<
+    EsQueryRuleParams<SearchType.esqlQuery>
+  >({
+    ...ruleParams,
+    timeWindowSize: timeWindowSize ?? DEFAULT_VALUES.TIME_WINDOW_SIZE,
+    timeWindowUnit: timeWindowUnit ?? DEFAULT_VALUES.TIME_WINDOW_UNIT,
+    // ESQL queries compare conditions within the ES query
+    // so only 'met' results are returned, therefore the threshold should always be 0
+    threshold: [0],
+    thresholdComparator: DEFAULT_VALUES.THRESHOLD_COMPARATOR,
+    size: DEFAULT_VALUES.SIZE,
+    esqlQuery: esqlQuery ?? { esql: '' },
+    aggType: DEFAULT_VALUES.AGGREGATION_TYPE,
+    groupBy: DEFAULT_VALUES.GROUP_BY,
+    termSize: DEFAULT_VALUES.TERM_SIZE,
+    searchType: SearchType.esqlQuery,
+  });
```

Comment:
**question:** if someone uses the HTTP API directly to create ESQL rules. Are any of these fields something we'd want them to prevent changing? Ex: `threshold`, `thresholdComparator`, etc. If so, should we update the backend validation accordingly?

Reply:
Resolved in this commit https://github.com/elastic/kibana/pull/164073/commits/cf0b33fcbd05d60910a01acbe96332f8dec5b0d1